DbAdapterTestUtilities/Stubs/StubPrimaryKeyValue.cpp

```diff
@@ -74,33 +74,34 @@ namespace systelab { namespace db { namespace test_utility {
 
 	unsigned int StubPrimaryKeyValue::getFieldValuesCount() const
 	{
-		return (unsigned int) m_fieldValues.size();
+		return static_cast<unsigned int>(m_fieldValues.size());
 	}
 
 	IFieldValue& StubPrimaryKeyValue::getFieldValue(unsigned int index) const
 	{
-		if (index < m_fieldValues.size())
-		{
-			return *(m_fieldValues[index]);
-		}
-		else
+		if (index >= m_fieldValues.size())
 		{
-			throw std::runtime_error( "Invalid primary key field index" );
+			throw std::runtime_error("Invalid primary key field index");
+
 		}
+
+		return *(m_fieldValues.at(index));
 	}
 
 	IFieldValue& StubPrimaryKeyValue::getFieldValue(const std::string& fieldName) const
 	{
-		unsigned int nFields = (unsigned int) m_fieldValues.size();
-		for (unsigned int i = 0; i < nFields; i++)
-		{
-			if (m_fieldValues[i]->getField().getName() == fieldName)
+		const auto fieldValue = std::find_if(m_fieldValues.cbegin(), m_fieldValues.cend(),
```

Comment:
Since you are now in C++20 you could try the ranges library:
```
const auto fieldValue = std::ranges::find_if(m_fieldValues, [&fieldName](const auto& field){ return field->getField().getName() == fieldName});
```
All the container-related algorithms have an equivalent under `ranges` namespace that get _an iterable_ (something with `.begin()` and `.end()`) and he does the work, simplifying your code.

Reply:
Looks nice this new standard addition, I'll change it :)